File: leetcode073.py
class Solution:
    def setZeroes(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        x = []
        y = []
        l1,l2 = len(matrix), len(matrix[0])
        for i in range(l1):
            for j in range(l2):
                if matrix[i][j] == 0:
                    if i not in x:
                        x.append(i)
                    if j not in y:
                        y.append(j)
        for t in x:
            matrix[t][:] = [0 for _ in range(l2)]
        for t in y:
            for j in range(l1):
                matrix[j][t] = 0

File: leetcodec++/leetcode073.cpp
//
// Created by 黄敏辉 on 2019-07-19.
//

class Solution {
public:
    const int LetMeBe = -65435643;
    void setZeroes(vector<vector<int>>& matrix) {
        for (int i(0); i < matrix.size(); i ++){
            for(int j(0); j < matrix[0].size(); j++){
                if (matrix[i][j] == 0)
                    matrix[i][j] = LetMeBe;
            }
        }
        for (int i(0); i < matrix.size(); i ++){
            for(int j(0); j < matrix[0].size(); j++){
                if (matrix[i][j] == LetMeBe) {
                    helper(matrix, i, j);
                    matrix[i][j] = 0;
                }
            }
        }
    }

    void helper(vector<vector<int>>& matrix, int i, int j){
        for (int k(0);  k < matrix[0].size(); k++){
            if (matrix[i][k] != LetMeBe)
                matrix[i][k] = 0;
        }
        for (int k(0); k < matrix.size(); k ++){
            if(matrix[k][j] != LetMeBe)
                matrix[k][j] = 0;
        }

    }
};
